src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt

```diff
@@ -0,0 +1,529 @@
+@file:Suppress("ClassName")
+
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.activities
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.web.reactive.server.WebTestClient
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders.NomisDataBuilder
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders.Repository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderProgramProfile
+import java.time.LocalDate
+
+class GetAllocationResourceIntTest : IntegrationTestBase() {
+
+  @Autowired
+  private lateinit var repository: Repository
+
+  @Autowired
+  private lateinit var nomisDataBuilder: NomisDataBuilder
+
+  private val today = LocalDate.now()
+  private val yesterday = today.minusDays(1)
+  private val tomorrow = today.plusDays(1)
+
+  @BeforeEach
+  fun setup() {
+    nomisDataBuilder.build {
+      programService()
+    }
+  }
+
+  @AfterEach
+  fun cleanUp() {
+    repository.deleteOffenders()
+    repository.deleteActivities()
+    repository.deleteProgramServices()
+  }
+
+  @Nested
+  @DisplayName("GET /allocations/ids")
+  inner class FindActiveAllocations {
+
+    @Nested
+    inner class Api {
+      @Test
+      fun `access forbidden when no authority`() {
+        webTestClient.get().uri("/allocations/ids?prisonId=BXI")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.get().uri("/allocations/ids?prisonId=BXI")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.get().uri("/allocations/ids?prisonId=BXI")
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `invalid prison should return not found`() {
+        webTestClient.get().uri("/allocations/ids?prisonId=XXX")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+          .exchange()
+          .expectStatus().isBadRequest
+          .expectBody()
+          .jsonPath("userMessage").value<String> {
+            assertThat(it).contains("Prison with id=XXX does not exist")
+          }
+      }
+    }
+
+    @Nested
+    inner class Paging {
```

Comment:
The new course allocation finder paging tests are almost identical to the course activity tests - again how can we stop these diverging?